commands/src/test/java/discord4j/commands/example/SimpleCommandProvider.java

```diff
@@ -0,0 +1,27 @@
+package discord4j.commands.example;
+
+import discord4j.commands.BaseCommand;
+import discord4j.commands.CommandProvider;
+import discord4j.core.event.domain.message.MessageCreateEvent;
+
+import java.util.Optional;
+
+public class SimpleCommandProvider implements CommandProvider {
+
+    @Override
+    public Optional<? extends BaseCommand> provide(MessageCreateEvent context) {
+        if (context.getMessage().getContent().isPresent()) {
```

Comment:
You could use filter/map here on the Message#getContent() optional and just return that chain